File: Data Structures & Algorithms/add-two-numbers/submission-0.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */

class Solution {
public:
    ListNode* addTwoNumbers(ListNode* l1, ListNode* l2) {
        ListNode dummy(0);
        ListNode* out=&dummy;
        int carry=0;
        while ((l1 != nullptr || l2 != nullptr) || carry != 0){
            int a=(l1)? l1->val:0;
            int b=(l2)? l2->val:0;
            int sum=a+b+carry;

            int tmp= sum%10;
            carry=(int)sum/10;
            out->next=new ListNode(tmp);
            out=out->next;
            // if(carry!=0 && l1->next==nullptr){
            //     out->next=new ListNode(carry);
            //     out=out->next;
            //     carry=0;
            // }
            if(l1) l1=l1->next;
            if(l2) l2=l2->next;
        }
        return dummy.next;
    }
};

File: Data Structures & Algorithms/reorder-linked-list/submission-0.js
/**
 * Definition for singly-linked list.
 * class ListNode {
 *     constructor(val = 0, next = null) {
 *         this.val = val;
 *         this.next = next;
 *     }
 * }
 */

class Solution {
    /**
     * @param {ListNode} head
     * @return {void}
     */
    reorderList(head) {
        let fp=head.next,sp=head;
        while(fp!==null && fp.next!==null){
            sp=sp.next;
            fp=fp.next.next;
        }
        let sec= sp.next;
        let prev=(sp.next=null);

        while(sec!==null){
            let tmp=sec.next;
            sec.next=prev;
            prev=sec;
            sec=tmp;
        }   

        let first=head;
        sec=prev;
        while(sec!==null){
            const tmp1=first.next;
            const tmp2=sec.next;
            first.next=sec;
            sec.next=tmp1;
            first=tmp1;
            sec=tmp2;
        }
    }
}
